src/internal/storage/server.go

```diff
@@ -220,23 +225,48 @@ func (w *writer) Write(data []byte) (int, error) {
 
 func (s *Server) ReadFilesetCDR(request *storage.ReadFilesetRequest, server storage.Fileset_ReadFilesetCDRServer) error {
 	ctx := server.Context()
-	// TODO: Use task chain here.
-	return s.readFileset(ctx, request, func(f fileset.File) error {
+	cache, err := lru.New[string, string](128)
+	if err != nil {
+		panic(err)
+	}
+	taskChain := taskchain.New(ctx, semaphore.NewWeighted(int64(taskParallelism)))
+	if err := s.readFileset(ctx, request, func(f fileset.File) error {
 		var refs []*cdr.Ref
-		for _, dr := range f.Index().File.DataRefs {
-			ref, err := s.Chunks.CDRFromDataRef(ctx, dr)
-			if err != nil {
+		refNum := len(f.Index().File.DataRefs)
```

Comment:
[nit]: refNum implies that each reference has a number (i.e. an id), whereas `numRefs` implies a count of refs.
I'd suggest renaming this to one of `numRefs`, `countRefs`, `refCount`.